From 40ff5fca2e026c238ec1b4160165872c2711acce Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Mon, 16 Nov 2020 22:21:35 -0500
Subject: [PATCH 679/767] drivers: marvell: otx2-sdei-ghes: fix BERT init error
 when RAS is disabled

commit 9b7e531bc10d1ee5c8d4cd3ea0826bb1887d5c3f from
git@git.assembla.com:cavium/WindRiver.linux.git

If the firmware disables RAS support at runtime, do not report
a BERT error during initialization.  Simply log an informative
message and continue.

Change-Id: I82ec8bc9941073b1bb585b5e7f2b8bfc92b9745a
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/40192
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
---
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
index b65e346400ec..401c860a65e4 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
@@ -649,8 +649,13 @@ static int __init sdei_ghes_bert_init(struct device_node *of_node)
 
 	bed_ring = memblock_va;
 
+	/*
+	 * This indicates that the firmware does not support BERT.
+	 * This is not an error, exit with success.
+	 */
 	if (!bed_ring->size) {
-		pr_err("Invalid ring size %d\n", bed_ring->size);
+		pr_info("BERT support disabled by firmware\n");
+		ret = 0;
 		goto exit;
 	}
 
-- 
2.31.1

